nixos/lib/default.nix

```diff
@@ -0,0 +1,46 @@
+let
```

Comment:
Are you sure this is the right place for evaluating NixOS? I'd have expected this to be in `nixos/default.nix` (but I know that place is already used). Maybe someone else has a good idea but this location makes it seem like importing that file gives me library functions around NixOS and not NixOS

Reply:
It is not great, I admit, and as you've noted that's mostly due to `nixos/default.nix` being taken. I've considered turning the existing `nixos/default.nix` into a `__functor`, but I've seen the command line tools behave weird around `__functor` and I don't want to touch such an essential file for an experimental feature.

I've also considered `nixos/eval-config.nix`. While familiarity is good, it is too easily confused with `nixos/lib/eval-config.nix`, which behaves rather differently, both wrt module list and a ton of legacy behavior we don't need.

Reply:
What about `nixos/minimal-eval.nix`?

Reply:
`nixos/minimal-eval.nix` does sound less confusing. or maybe `nixos/lib/minimal-eval.nix`? anything that doesn't break the pattern of `import ./lib` pulling in all lib functions, really.

Reply:
> anything that doesn't break the pattern of `import ./lib` pulling in all lib functions

Actually, that's exactly what this is doing, when looking at the `flake.nix` change. Its `lib.nixos` is simply this file.
So then it *should* be `lib/default.nix` and the fact that the functions are about minimal evaluation is currently a coincidence.

Reply:
from the angle of what the flake exports you're totally right. but there's a lot of other stuff in `nixos/lib` (like that now-standard eval-config.nix) that isn't available from `import ./nixos/lib`. so i guess it's fine either way, and we retract the suggestion.

Reply:
Right, I now better understand your suggestion. Depending on whether those are intended for "public" consumption, the functions in those other files can later be exposed through `nixos/lib/default.nix` or moved to a `nixos/common` directory where exposure is not desirable.

Reply:
we have `nixos/lib/eval-config.nix`, so I would propose `nixos/lib/eval-config-minimal.nix`. Also `nixos/lib/eval-config.nix` should be using `nixos/lib/eval-config-minimal.nix`, so *prove* it is minimal and therefore the status quo cna be built on top of it.

Reply:
I've moved the implementation as @Ericson2314 suggested, kept `nixos/lib/default.nix` for the purpose of being `lib.nixos` in the flake and anyone who likes the `lib` style.

I'm undecided whether it only proves _a point_ or whether it should be a requirement. If we find a reason to disconnect the two, proving a point shouldn't stop us from doing that. I can imagine that we find a useful change along the way, that we don't want to apply to conventional NixOS, yet I do hope the two can remain as similar as possible.

Reply:
Yeah if we have need to direct them, sure. I am not yet familiar with the more ambitious plans discussed before.